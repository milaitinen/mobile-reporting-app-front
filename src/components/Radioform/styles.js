import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({

    $lightBlue: '$inactive',
    $gray: '$disabledPlaceholder',
    $blue: '$active',

    radioInputContainer: {
        height: 40,
        marginBottom: 10,
        paddingHorizontal: 1,
        marginHorizontal: 0,
        alignSelf: 'stretch',
        alignItems: 'center',
        justifyContent: 'flex-start',
        borderRadius: '$inputBorderRadius',
        backgroundColor: '$primaryWhite',
    },

    selectedInputContainer: {
        height: 40,
        marginBottom: 10,
        paddingHorizontal: 1,
        marginHorizontal: 0,
        alignSelf: 'stretch',
        alignItems: 'center',
        justifyContent: 'flex-start',
        borderRadius: '$inputBorderRadius',
        backgroundColor: '$primaryWhite',
        borderWidth: 2,
        borderColor: '$active'
    },

    radioLabel: {
        fontFamily: '$primaryFont',
        color: '$placeholder',
        textAlignVertical: 'center',
        textAlign: 'left',
        paddingBottom: 3,
        marginRight: 10
    },

    buttonWrap: {
        height: 38,
        paddingLeft: 9,
        justifyContent: 'center'
    },

    labelWrap: {
        height: 38,
        flex: 1,
    },

    selectedLabel: {
        fontFamily: '$primaryFont',
        color: '$active',
        textAlignVertical: 'center',
        textAlign: 'left',
        paddingBottom: 3,
        marginRight: 10,
    },

    disabled: {
        backgroundColor: '$disabled',
        borderColor: '$disabledBorder',
        borderWidth: 2,
    },

    disabledText: {
        color: '$disabledPlaceholder',
        fontFamily: '$primaryFont',
    }

});

export default styles;