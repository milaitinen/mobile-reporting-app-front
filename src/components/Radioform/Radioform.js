import RadioForm, { RadioButton, RadioButtonInput, RadioButtonLabel } from 'react-native-simple-radio-button';
import React, { Component } from 'react';
import styles from './styles';

class Radioform extends Component {
    constructor(props) {
        super(props);
        this.state = {
            index: this.props.initial || 0
        };
    }

    press(index) {
        this.setState({ index: index });
    }

    render() {
        const editable = this.props.editable;
        const isPreview = this.props.isPreview;
        const container = (editable || !isPreview) ? styles.radioInputContainer : [styles.radioInputContainer, styles.disabled];
        const button = (editable || !isPreview) ? styles.$lightBlue : styles.$gray;
        const label = (editable || !isPreview) ? styles.radioLabel : styles.disabledText;
        return (
            <RadioForm>
                {this.props.options.map((obj, i) =>
                    <RadioButton
                        key={i}
                        style={this.state.index === i ? styles.selectedInputContainer : container}>
                        <RadioButtonInput
                            disabled={!editable}
                            obj={obj}
                            index={i}
                            isSelected={this.state.index === i}
                            onPress={(value) => {
                                this.props.onPress(value);
                                this.press(i);
                            }}
                            borderWidth={1.5}
                            buttonInnerColor={styles.$blue}
                            buttonOuterColor={this.state.index === i ? styles.$blue : button}
                            buttonSize={16}
                            buttonOuterSize={24}
                            buttonWrapStyle={styles.buttonWrap}
                        />
                        <RadioButtonLabel
                            disabled={!editable}
                            obj={obj}
                            index={i}
                            labelHorizontal={true}
                            onPress={(value) => {
                                this.props.onPress(value);
                                this.press(i);
                            }}
                            labelStyle={this.state.index === i ? styles.selectedLabel : label}
                            labelWrapStyle={styles.labelWrap}
                        />
                    </RadioButton>
                )}
            </RadioForm>
        );
    }
}

export default Radioform;