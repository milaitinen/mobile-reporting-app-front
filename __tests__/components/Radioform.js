import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import Radioform from '../../src/components/Radioform/Radioform';

configure({ adapter: new Adapter() });

const options = [
    {
        label: 'Option 1',
        value: {
            field_option_id: 1,
            field_id: 3,
            value: 'Option 1'
        }
    },
    {
        label: 'Option 2',
        value: {
            field_option_id: 2,
            field_id: 3,
            value: 'Option 2'
        }
    }
];

describe('Radioform component', () => {
    const radioform = renderer.create(<Radioform
        options={options}
        editable={true}
        initial={0}
        itemRealKey='value'
        onPress={jest.fn()}
    />);

    const inst = radioform.getInstance();

    it('renders correctly', () => {
        expect(radioform.toJSON()).toMatchSnapshot();
    });

    it('works as expected when press function called', () => {
        inst.press(2);
        expect(inst.state.index).toEqual(2);
    });
});

describe('Radioform presses', () => {
    const spy = sinon.spy(Radioform.prototype, 'press');
    const radioform = shallow(
        <Radioform
            options={options}
            editable={true}
            initial={0}
            itemRealKey='value'
            onPress={jest.fn()}
        />
    );

    it('work as expected when radio button pressed', () => {
        radioform.find('RadioButtonInput').forEach(child =>{
            child.simulate('press');
            expect(spy.called).toEqual(true);
        });
    });

    it('work as expected when radio label pressed', () => {
        radioform.find('RadioButtonLabel').forEach(child =>{
            child.simulate('press');
            expect(spy.called).toEqual(true);
        });
    });
});

