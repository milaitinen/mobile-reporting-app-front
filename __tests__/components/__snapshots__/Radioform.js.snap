// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radioform component renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      false,
    ]
  }
>
  <View
    style={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        style={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "borderColor": "#2196f3",
              "borderRadius": 12,
              "borderWidth": 1.5,
              "height": 24,
              "justifyContent": "center",
              "opacity": 1,
              "width": 24,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 10,
                },
                Object {
                  "backgroundColor": "#2196f3",
                  "height": 20,
                  "width": 20,
                },
                Object {
                  "borderRadius": 8,
                  "height": 16,
                  "width": 16,
                },
                Object {
                  "backgroundColor": "#2196f3",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              false,
              Object {
                "color": undefined,
              },
              undefined,
            ]
          }
        >
          Option 1
        </Text>
      </View>
    </View>
  </View>
  <View
    style={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        style={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "borderColor": "#2196f3",
              "borderRadius": 12,
              "borderWidth": 1.5,
              "height": 24,
              "justifyContent": "center",
              "opacity": 1,
              "width": 24,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 10,
                },
                false,
                false,
                false,
              ]
            }
          />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              false,
              Object {
                "color": undefined,
              },
              undefined,
            ]
          }
        >
          Option 2
        </Text>
      </View>
    </View>
  </View>
</View>
`;
